### ISSUE TITLE
Feature: pass `editorState` to `ElementNode.getChildren`

### ISSUE THREAD

**Issue by jason89521** [facebook/lexical#5222]
## Description

I am currently in the process of implementing a custom image plugin. In contrast to the one in the playground, which directly inserts an image into the editor, my plugin requires the image to be uploaded first. The image node should also have a `progress` field to display a progress bar, as shown in the following video:

https://github.com/facebook/lexical/assets/47748132/706f3b11-99d7-4239-88f8-dc9585c8dc51

However, the upload state is being pushed to the history's undo stack. I have managed to address this issue by using the following code snippet:

```ts
editor.update(() => {
  // my code
}, { tag: 'history-merge' })
```

Nevertheless, this approach only works when the editor doesn't update during the uploading process. To ensure that the undo/redo stack reflects the latest uploading progress, I need to traverse all the nodes inside the old root node. Unfortunately, the `getChildren` method currently returns the latest nodes.

To achieve my goal, I have written the following code:

```ts
export function $getChildrenSnapshot(
  node: ElementNode,
  editorState: EditorState,
): LexicalNode[] {
  const children: LexicalNode[] = [];
  let child: LexicalNode | null = isNull(node.__first)
    ? null
    : $getNodeByKey(node.__first);
  while (isNonNullable(child)) {
    children.push(child);
    child = isNull(child.__next)
      ? null
      : $getNodeByKey(child.__next, editorState);
  }

  return children;
}
```

I am wondering if it would be possible to allow passing the `editorState` to the `getChildren` method. I am willing to help implement this feature.

## TL;DR

I am seeking permission to pass `editorState` to the `ElementNode.getChildren` method to enhance functionality in my custom image plugin.

**Comment by acywatson** [comment]
I don't think this is the right approach. Instead, I would try using a decorator node and updating the upload state in there without touching the Lexical state.

**Comment by jason89521** [comment]
> I don't think this is the right approach. Instead, I would try using a decorator node and updating the upload state in there without touching the Lexical state.

Are you saying to write another decorator node to update the uploading state of the image node? I'm a bit unclear on how to proceed with that.
